pisa/core/pipeline.py

```diff
@@ -372,16 +369,18 @@ def get_outputs(self, **get_outputs_kwargs):
         else:
             outputs = self._get_outputs(**get_outputs_kwargs)
         return outputs
-        
+
     def _get_outputs(self, output_binning=None, output_key=None):
         """Get MapSet output"""
 
         self.run()
 
         if output_binning is None:
             output_binning = self.output_binning
-        elif isinstance(output_binning, VarBinning):
-            self.check_VarBinning(output_binning)
+        if isinstance(output_binning, VarBinning):
+            # checks also have to be done when no new output_binning is passed
```

Comment:
No, they shouldn't. Especially the `assert_exclusive_varbinning` test can be a bit of an computational effort if you have a lot of MC. Running this every time the `_get_outputs` function is called (which happens a lot during a fit) might slow down the fit quite a bit. If `self.output_binning` is used, it was already tested, either in the init or in its setter, so there is no need to test it again. Only if the binning changes (by setting it in this function) we need to test it.

Reply:
resolved (check if exclusive now only upon external output_binning, but events apply_modes always)